### PR TITLE
[Merged by Bors] - feat(MorphismProperty/Presheaf): add `relative` morphism property

### PR DESCRIPTION
**Pull request by callesonne**
Given a morphism property P and a functor F, we define the morphism property `P.relative` of morphisms in the codomain of `F`. This is defined as being relatively representable with respect to `F`, and that any represented pullback of the morphism satisfies P.

Co-authored-by: Joël Riou <rioujoel@gmail.com>

---

- [x] depends on: #16140
- [x] depends on: #16141

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by joelriou** [comment]
(I have contributed to this PR, but I just would like to say that this looks good to me.)

**Review by dagurtomas — APPROVED** [review]

File: Mathlib/CategoryTheory/MorphismProperty/Representable.lean
Comment (callesonne) [reply Mathlib/CategoryTheory/MorphismProperty/Representable.lean]: Great suggestion, fixed!

File: Mathlib/CategoryTheory/MorphismProperty/Representable.lean
Comment (erdOne) [inline Mathlib/CategoryTheory/MorphismProperty/Representable.lean:318]:
```suggestion
abbrev presheaf : MorphismProperty (Dᵒᵖ ⥤ Type v₂) := P.relative yoneda
```
Not tested

File: Mathlib/CategoryTheory/MorphismProperty/Representable.lean
Comment (erdOne) [inline Mathlib/CategoryTheory/MorphismProperty/Representable.lean:305]: I feel that the name is a bit too generic but I don't have a better name.

File: Mathlib/CategoryTheory/MorphismProperty/Representable.lean
Comment (erdOne) [inline Mathlib/CategoryTheory/MorphismProperty/Representable.lean:343]: I think `hf` is not needed here.

File: Mathlib/CategoryTheory/MorphismProperty/Representable.lean
Comment (callesonne) [reply Mathlib/CategoryTheory/MorphismProperty/Representable.lean:318]: Fixed!

File: Mathlib/CategoryTheory/MorphismProperty/Representable.lean
Comment (callesonne) [reply Mathlib/CategoryTheory/MorphismProperty/Representable.lean:343]: Nice spot!

File: Mathlib/CategoryTheory/MorphismProperty/Representable.lean
Comment (callesonne) [reply Mathlib/CategoryTheory/MorphismProperty/Representable.lean:305]: I agree that its generic, but I also don't have a better name for it.

**Comment by erdOne** [comment]
LGTM modulo the comment on naming. Maybe merge it if there aren't better names.
maintainer merge

File: Mathlib/CategoryTheory/MorphismProperty/Representable.lean
Comment (kbuzzard) [reply Mathlib/CategoryTheory/MorphismProperty/Representable.lean:305]: It's stuck away in an obscure namespace so I think this is fine.

**Comment by kbuzzard** [comment]
Great to see progress here :-) There are so many functors we need to represent!

bors merge

**Pull request closed without merging**
